_posts/2023-10-23-unidirectional-topic-operator.md

```diff
@@ -0,0 +1,310 @@
+    strimzi.io/cluster: my-cluster
+spec:
+  partitions: 1
+  replicas: 1
+  config:
+    min.insync.replicas: 1" | kubectl create -f -
+kafkatopic.kafka.strimzi.io/uto-test created
+
+$ kubectl get kt uto-test -o jsonpath='{.status}'
+{"conditions":[{"lastTransitionTime":"2023-10-23T08:41:53.501883404Z","status":"True","type":"Ready"}],"observedGeneration":1,"topicName":"uto-test"}
+
+$ kubectl exec my-cluster-kafka-0 -- bin/kafka-topics.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --describe --topic uto-test
+Topic: uto-test	TopicId: gWrR6uDrSNSSfu_ubGndCg	PartitionCount: 1	ReplicationFactor: 1	Configs: min.insync.replicas=1,message.format.version=3.0-IV1
+	Topic: uto-test	Partition: 0	Leader: 1	Replicas: 1	Isr: 1
+	
+$ kubectl delete kt uto-test
+kafkatopic.kafka.strimzi.io "uto-test" deleted
+```
+
+Optionally, you can delete the internal topics created by the BTO, which are not used anymore.
+
+```sh
+$ kubectl delete $(kubectl get kt -o name | grep strimzi-store-topic)
+kafkatopic.kafka.strimzi.io "strimzi-store-topic---effb8e3e057afce1ecf67c3f5d8e4e3ff177fc55" deleted
+
+$ kubectl delete $(kubectl get kt -o name | grep strimzi-topic-operator)
+kafkatopic.kafka.strimzi.io "strimzi-topic-operator-kstreams-topic-store-changelog---b75e702040b99be8a9263134de3507fc0cc4017b" deleted
+```
+
+Finally, we can also choose to unmanage the Kafka internal topics, without deleting them inside the Kafka cluster.
+Here I'm just showing the `__consumer_offsets`, but the procedure is the same for the other internal topics.
+
+```sh
+$ kubectl annotate $(kubectl get kt -o name | grep consumer-offsets) strimzi.io/managed="false"
+kafkatopic.kafka.strimzi.io/consumer-offsets---84e7a678d08f4bd226872e5cdd4eb527fadc1c6a annotate
+
+$ kubectl delete $(kubectl get kt -o name | grep consumer-offsets)
+kafkatopic.kafka.strimzi.io "consumer-offsets---84e7a678d08f4bd226872e5cdd4eb527fadc1c6a" deleted
+
+$ kubectl exec my-cluster-kafka-0 -- bin/kafka-topics.sh --bootstrap-server my-cluster-kafka-bootstrap:9092 --describe --topic __consumer_offsets
+Topic: __consumer_offsets	TopicId: AZQjCm_qSYyjSxRYNPsU5A	PartitionCount: 50	ReplicationFactor: 3	Configs: compression.type=producer,min.insync.replicas=2,cleanup.policy=compact,segment.bytes=104857600,message.format.version=3.0-IV1
+	Topic: __consumer_offsets	Partition: 0	Leader: 2	Replicas: 2,1,0	Isr: 2,1,0
+	Topic: __consumer_offsets	Partition: 1	Leader: 1	Replicas: 1,0,2	Isr: 1,0,2
+	Topic: __consumer_offsets	Partition: 2	Leader: 0	Replicas: 0,2,1	Isr: 0,2,1
+	...
+```
+
+### Downgrading from UTO to BTO
+
+Downgrading to BTO is as easy as disabling the feature gate.
+
+**Note that it may take some time to rebuild the internal state store if your Kafka cluster hosts many topics.**
+**Make sure to have adequate CPU and memory resources for that.**
+
+```sh
+$ kubectl set env deploy strimzi-cluster-operator STRIMZI_FEATURE_GATES="-UnidirectionTopicOperator"
+deployment.apps/strimzi-cluster-operator env updated
```

Comment:
Again, maybe we want to bring this up to sit under _Downgrading to BTO is as easy as disabling the feature gate._

Reply:
I think it's better to warn users before the command.